apps/condo/domains/organization/schema/FindOrganizationsByTinService.js

```diff
@@ -0,0 +1,145 @@
+ */
+const ERRORS = {
+    EMPTY_TIN: {
+        query: 'findOrganizationsByTin',
+        variable: ['data', 'tin'],
+        code: BAD_USER_INPUT,
+        type: 'EMPTY_TIN',
+        message: 'Empty tin',
+    },
+    DV_VERSION_MISMATCH: {
+        ...COMMON_ERRORS.DV_VERSION_MISMATCH,
+        query: 'findOrganizationsByTin',
+    },
+    WRONG_SENDER_FORMAT: {
+        ...COMMON_ERRORS.WRONG_SENDER_FORMAT,
+        query: 'findOrganizationsByTin',
+    },
+}
+
+const FindOrganizationsByTinService = new GQLCustomSchema('FindOrganizationsByTinService', {
+    types: [
+        {
+            access: true,
+            type: 'input FindOrganizationsByTinInput { dv: Int!, sender: JSON!, tin: String! }',
+        },
+        {
+            access: true,
+            type: 'type FindOrganizationsByTinOrganizationType { id: String!, name: String! }',
+        },
+        {
+            access: true,
+            type: 'type FindOrganizationsByTinOutput { organizations: [FindOrganizationsByTinOrganizationType]! }',
+        },
+    ],
+
+    queries: [
+        {
+            access: access.canFindOrganizationsByTin,
+            schema: 'findOrganizationsByTin (data: FindOrganizationsByTinInput!): FindOrganizationsByTinOutput',
+            doc: {
+                summary: 'Returns all organizations by TIN. Used to get organizations that the user can apply to join',
+                errors: ERRORS,
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+                const { tin, dv, sender } = data
+                const authedItemId = get(context, 'authedItem.id', null)
+                if (!authedItemId) throw new Error('no authedItemId!')
+
+                checkDvAndSender(data, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
+
+                if (!tin) throw new GQLError(ERRORS.EMPTY_TIN, context)
+
+                // NOTE: we don't use "isFeatureEnabled" because it can skip the request limit for all users
+                const userWhiteList = await featureToggleManager.getFeatureValue(context, USER_WHITE_LIST_FOR_FIND_ORGANIZATIONS_BY_TIN, [], { user: authedItemId }) || []
+                const skipRequestLimit = get(context, 'authedItem.isAdmin', false)
+                    || get(context, 'authedItem.isSupport', false)
+                    || (Array.isArray(userWhiteList) && userWhiteList.includes(authedItemId))
+                if (!skipRequestLimit) {
+                    await checkDailyRequestLimitCountersByUser(context, 'findOrganizationsByTin', authedItemId)
+                    await checkTotalRequestLimitCountersByUser(context, 'findOrganizationsByTin', authedItemId, MAX_TOTAL_REQUESTS)
+                } else {
+                    logger.info({ msg: `User "${authedItemId}" skip request limit` })
+                }
+
+                await FindOrganizationsByTinLog.create(context, {
```

Comment:
i'm not sure that we really need to add separate model for logs, our log sysyem not doesn't fit because logs ttl?

Reply:
this way we can analyze who requested what and when over a long period of time